### PR TITLE
fix: stop flashing Tildy assistant name when loading custom assistants

### PR DESCRIPTION
**Pull request by thedadams**

### CODE REVIEW

File: app/page.tsx [hunk 1]
```diff
@@ -6,13 +6,12 @@ import Chat from '@/components/chat';
...
 import { ChatContextProvider } from '@/contexts/chat';
 import { NavContext } from '@/contexts/nav';
-import { tildy } from '@/config/assistant';
 import ExploreModal from '@/components/explore/ExploreModal';
 import { useDisclosure } from '@nextui-org/react';
 
 function RunFile() {
   const [script, _setScript] = useState<string>(
-    useSearchParams().get('file') ?? tildy
+    useSearchParams().get('file') ?? ''
```
Comment (thedadams) [inline app/page.tsx:14]: Changing this is really all I wanted to do. However, this caused a bunch of errors to start popping up on launch, staring with the `getMe` stuff.

Most of the other changes are resolving the errors that popped up as a result.

**Review by njhale — APPROVED** [review]

File: contexts/chat.tsx [hunk 2]
```diff
@@ -336,7 +334,7 @@ const ChatContextProvider: React.FC<ChatContextProps> = ({
       setScript(script);
       setThread(newThread.meta.id);
       setSelectedThreadId(newThread.meta.id);
-      setWorkspaceDir(newThread.meta.workspace);
+      setWorkspaceDir(newThread.meta.workspace).then();
```
Comment (StrongMonkey) [inline contexts/chat.tsx:337]: Hmm what is this?

File: contexts/chat.tsx [hunk 3]
```diff
@@ -186,12 +181,14 @@ const ChatContextProvider: React.FC<ChatContextProps> = ({
         if ((initialScript && initialScriptId) || threads.length === 0) {
           // if both threads and scriptId are set, then always create a new thread
           const newThread = await createThread(
-            script,
-            scriptDisplayName ?? defaultScriptName,
+            script !== '' ? script : tildy,
```
Comment (StrongMonkey) [inline contexts/chat.tsx:184]: can be `script ? script : tildy`? Are you specifially checking undefined option?

**Review by StrongMonkey — APPROVED** [review]

File: contexts/chat.tsx (hunk 2 above)
Comment (thedadams) [reply contexts/chat.tsx:337]: Will remove. This was just a to get rid of a squiggly line that was bothering me.

File: contexts/chat.tsx (hunk 3 above)
Comment (thedadams) [reply contexts/chat.tsx:184]: I'll update this.

File: contexts/chat.tsx (hunk 2 above)
Comment (thedadams) [reply contexts/chat.tsx]: Done.

File: contexts/chat.tsx (hunk 3 above)
Comment (thedadams) [reply contexts/chat.tsx]: Done.

Comment (ryanhopperlowe) [reply contexts/chat.tsx]: I know it's already merged, but a better solution is `script || tildy`. in JavaScript `''` is a falsy value

**Pull request merged**